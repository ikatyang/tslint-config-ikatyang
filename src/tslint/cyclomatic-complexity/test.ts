// good

function good1() {
  console.log("do something");
}

// bad

function bad1() {
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
  if (true) { console.log("do something") }
}
