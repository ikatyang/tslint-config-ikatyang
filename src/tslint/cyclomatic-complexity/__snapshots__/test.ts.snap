// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint 1`] = `
// good

function good1() {
  console.log("do something");
}

// bad

function bad1() {
~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  if (true) { console.log("do something") }
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
}
~ [The function bad1 has a cyclomatic complexity of 21 which is higher than the threshold of 20]

`;
